pkg/network/protocols/http/monitor.go

```diff
@@ -160,6 +161,7 @@ func NewMonitor(c *config.Config, offsets []manager.ConstantEditor, sockFD *ebpf
 		processMonitor:  processMonitor,
 		http2Enabled:    c.EnableHTTP2Monitoring,
 		http2Statkeeper: http2Statkeeper,
+		httpTLSEnabled:  c.EnableHTTPSMonitoring || c.EnableHTTP2Monitoring || c.EnableJavaTLSSupport || c.EnableGoTLSSupport,
```

Comment:
should be only 
```
httpTLSEnabled: c.EnableHTTPSMonitoring
```

1. Why is `c.EnableHTTP2Monitoring` here?
2. both goTLS and javaTLS won't work if `c.EnableHTTPSMonitoring` is disabled.

Reply:
I cleanup and split in another [PR](https://github.com/DataDog/datadog-agent/pull/16540)